nifi-nar-bundles/nifi-druid-bundle/nifi-druid-processors/src/main/java/org/apache/nifi/processors/PutDruid.java

```diff
@@ -0,0 +1,206 @@
+            .name("FAIL")
+            .description("FlowFiles are routed to this relationship when they cannot be parsed")
+            .build();
+
+    public static final Relationship REL_DROPPED = new Relationship.Builder()
+            .name("DROPPED")
+            .description("FlowFiles are routed to this relationship when they are outside of the configured time window, timestamp format is invalid, ect...")
+            .build();
+
+    public void init(final ProcessorInitializationContext context){
+        List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(DRUID_TRANQUILITY_SERVICE);
+        this.properties = Collections.unmodifiableList(properties);
+
+        Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_DROPPED);
+        relationships.add(REL_FAIL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships(){
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+    
+    //Method breaks down incoming flow file and sends it to Druid Indexing servic
+    private void processFlowFile(ProcessContext context, ProcessSession session){
+    	 //Get handle on Druid Tranquility session
+    	 DruidTranquilityService tranquilityController = context.getProperty(DRUID_TRANQUILITY_SERVICE)
+                 .asControllerService(DruidTranquilityService.class);
+         Tranquilizer<Map<String,Object>> tranquilizer = tranquilityController.getTranquilizer();
+         
+         final FlowFile flowFile = session.get();
+         if (flowFile == null || flowFile.getSize() == 0) {
+             return;
+         }
+         
+         //Get data from flow file body
+         final byte[] buffer = new byte[(int) flowFile.getSize()];
+         session.read(flowFile, new InputStreamCallback() {
+             @Override
+             public void process(final InputStream in) throws IOException {
+                 StreamUtils.fillBuffer(in, buffer);
+             }
+         });
+         
+         
+         String contentString = new String(buffer, StandardCharsets.UTF_8);
+         Map<String,Object> contentMap = null;
+         
+         //Create payload array from flow file content, one element per line
+         String[] messageArray = contentString.split("\\R");
+         
+         //Convert each array element from JSON to HashMap and send to Druid
+         for(String message: messageArray){
+         	try {
+         		contentMap = new ObjectMapper().readValue(message, HashMap.class);
+         		//contentMap = new ObjectMapper().readValue(message, HashMap.class);
+         	} catch (JsonParseException e) {
+         		getLogger().error("Error parsing incoming message array in the flowfile body");
```

Comment:
if the error message is the same probalby better to do a multicatch

Reply:
Changed to IOException to merge combine all of the JSON parsing exceptions.